testsuite/gateway/gateway_api/route.py

```diff
@@ -73,17 +74,22 @@ def remove_all_hostnames(self):
         self.model.spec.hostnames = []
 
     @modify
-    def set_match(self, backend: "Httpbin", path_prefix: str = None):
-        """Limits HTTPRoute to a certain path"""
-        match = {}
-        if path_prefix:
-            match["path"] = {"value": path_prefix, "type": "PathPrefix"}
-        for rule in self.model.spec.rules:
-            for ref in rule.backendRefs:
-                if backend.reference["name"] == ref["name"]:
-                    rule["matches"] = [match]
-                    return
-        raise NameError("This backend is not assigned to this Route")
+    def add_rule(self, backend: "Httpbin", *route_matches: RouteMatch):
+        """Adds rule to the Route"""
+        rules = {"backendRefs": [backend.reference]}
+        matches = (
+            list(route_matches)
+            if route_matches
+            else [RouteMatch(path=PathMatch(type=MatchType.PATH_PREFIX, value="/"))]
+        )
```

Comment:
```suggestion
   matches = list(route_matches)
   if len(matches) == 0:
       matches.append(RouteMatch(path=PathMatch(type=MatchType.PATH_PREFIX, value="/"))
```

Reply:
I somehow agree with your suggestion.
But the thing to discuss here is that if no `route_matches` is used for the rule,
```
        - path:
            type: PathPrefix
            value: /
```
is used automatically. No need to specifically defined `RouteMatch(path=PathMatch(type=MatchType.PATH_PREFIX, value="/"))`

Reply:
in that case, the condition does not need to be there